lib/_http_incoming.js

```diff
@@ -63,6 +60,16 @@ util.inherits(IncomingMessage, Stream.Readable);
 
 exports.IncomingMessage = IncomingMessage;
 
+Object.defineProperty(IncomingMessage.prototype, 'client', {
+  configurable: true,
+  enumerable: true,
+  get: util.deprecate(function() {
+    return this.socket;
+  }, 'client is deprecated, use socket or connection instead'),
+  set: util.deprecate(function(val) {
+    this.socket = this.connection = val;
+  }, 'client is deprecated, use socket or connection instead')
```

Comment:
Wait, if the setter is changing `this.client`, wouldn't we have to change the underlying property name (e.g. to something like `this._client`) otherwise it would lead to some kind of loop (setting `this.client` inside a `this.client` setter)?